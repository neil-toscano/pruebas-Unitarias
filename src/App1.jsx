export const App=()=>{
    return (<h1>Holaa mundo jsx</h1>
    )
}
import React from 'react'

