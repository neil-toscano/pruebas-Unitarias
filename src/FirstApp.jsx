import { Fragment } from "react"
import PropTypes from 'prop-types';

const saludo=()=>{
    return "hola buenas, tardes";
}

export const FirstApp=({title, subtitle, name})=>{
   
    const newMessage='Neil';

    return (
        <Fragment>
            <h1> { newMessage } </h1>
            <h2>{ saludo() }</h2>
            <h3 data-testid='test-title'>
                {title}
            </h3>
            <h5>{ subtitle }</h5>
            <h5>{ subtitle }</h5>
            <h4>{ name }</h4>
        </Fragment>
    );

}

FirstApp.propTypes={
    title:PropTypes.string.isRequired,
    subtitle:PropTypes.string,
}

FirstApp.defaultProps={
    title: "no hay titulo",
    subtitle: 34,
    name: "Neil",
}