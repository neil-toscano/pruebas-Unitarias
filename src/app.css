html{
    background-color: rgb(65, 63, 63);
    color: white;
    font-family: Arial, Helvetica, sans-serif;
    font-size: 1.3rem;
    padding: 70px;
}