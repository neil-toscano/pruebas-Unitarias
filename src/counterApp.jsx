import PropTypes from 'prop-types';
import { Fragment, useState } from 'react';

export const CounterApp=({value})=>{
    
    const [counter,setCounter]=useState(value);

    const handleAdd = () => {
       // setCounter(counter+1);
        setCounter((c)=>c+1);
    }

    const handleSubstract = () => { 
        setCounter((c)=>c-1);
    }
    
    const handleReset = () => {
        setCounter(value);
    }

    return(
        <Fragment>
            <h1> Counter App </h1>
            <h2> { value } </h2>
            <h3> { counter } </h3>
            <button onClick={ () => { handleAdd() }} > +1 </button>
            <button onClick={ ()=>{ handleSubstract() } }> -1 </button>
            <button onClick={ ()=>{ handleReset() } }> reset </button>
        </Fragment>
    )
}

CounterApp.propTypes={
    value:PropTypes.number.isRequired
}