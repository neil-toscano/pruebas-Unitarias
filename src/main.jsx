import React from 'react';
import  ReactDOM  from 'react-dom/client';
import { App } from './App1';
import { FirstApp } from './FirstApp';
import './app.css'
import {CounterApp}   from './counterApp';
ReactDOM.createRoot(document.getElementById('root')).render(
    <React.StrictMode>
        {/* <App/> */}
        {/* <FirstApp title='Soy Toscanos' subtitle='curso React' /> */}
        <CounterApp value={ 24 }/>
    </React.StrictMode>
);