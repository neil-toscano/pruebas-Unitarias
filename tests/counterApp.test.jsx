import { fireEvent, render, screen } from "@testing-library/react";
import { CounterApp } from "../src/counterApp";

describe('pruebas CounterApp', () => {
    const valorCounter = 100;

    test('debe de hacer match con el snapshopt', () => { 
        const { container } = render(<CounterApp value={ valorCounter } />);
        expect(container).toMatchSnapshot();
    });
    
    test('debe de mostrar el valor inicial', () => { 
        render(<CounterApp value={ valorCounter } />);
        expect(screen.getAllByText(valorCounter).length).toBe(2);

    });
    
    test('debe de incrementar con el boton +1', () => { 
        render(<CounterApp value={ valorCounter } />);
        fireEvent.click(screen.getByText('+1'));
        expect(screen.getByText('101')).toBeTruthy();
        
    });
    
    test('debe de decrementar con el boton -1', () => { 
        render(<CounterApp value={ valorCounter } />);
        fireEvent.click(screen.getByText('-1'));
        expect(screen.getByText('99')).toBeTruthy();
        
    });
    
    test('debe de resetear el estado', () => { 
        render(<CounterApp value={ 199 } />);
        fireEvent.click(screen.getByText('+1'));
        screen.debug();
        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByText('+1'));
        fireEvent.click(screen.getByText('reset'));
        expect(screen.getAllByText('199').length).toBe(2)

    });


});