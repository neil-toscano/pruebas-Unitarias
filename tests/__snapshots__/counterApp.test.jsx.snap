// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas CounterApp debe de hacer match con el snapshopt 1`] = `
<div>
  <h1>
     Counter App 
  </h1>
  <h2>
     
    100
     
  </h2>
  <h3>
     
    100
     
  </h3>
  <button>
     +1 
  </button>
  <button>
     -1 
  </button>
  <button>
     reset 
  </button>
</div>
`;
